misc/setup_thanos_storage.py

```diff
@@ -0,0 +1,117 @@
+import json
+import base64
+
+from kubernetes import client, config
+import boto3
+import yaml
+import argparse
+
+
+# 1. Create S3 bucket for Thanos
+def create_s3_bucket(bucket_name):
+    s3_client = boto3.client('s3')
+    s3_client.create_bucket(Bucket=bucket_name)
+
+# 2. Create IAM policy to access the bucket
+def create_policy(policy_name):
+    # Create the IAM policy document
+    policy_document = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Action": [
+                    "s3:PutObject",
+                    "s3:GetObject",
+                    "s3:AbortMultipartUpload",
+                    "s3:ListBucket",
+                    "s3:DeleteObject",
+                    "s3:ListMultipartUploadParts"
+                ],
+                "Resource": [
+                    "arn:aws:s3:::plantd-thanos",
+                    "arn:aws:s3:::plantd-thanos/*"
```

Comment:
The bucket name is hard-coded here. Should use the variable.

Reply:
fixed it